utils/resource-monitor/graph_pidstat.py

```diff
@@ -0,0 +1,215 @@
+    default='/tmp/hapi-performance/',
+    help='Provide the output path for the graphs and csv files.')
+
+def get_machine_mem():
+    """
+    Gets the RAM of the local machine.
+
+    Args:
+      None
+
+    Returns:
+      Integer representation of the local machine RAM
+    """
+    mem_info = subprocess.run(['grep', 'MemTotal', '/proc/meminfo'], stdout=subprocess.PIPE).stdout.split()
+    return int(int(mem_info[1])/1000000)
+
+def get_machine_cpu():
+    """
+    Gets the CPU/number of cores of the local machine.
+
+    Args:
+      None
+
+    Returns:
+      Integer representation of the number of cores of the local machine
+    """
+    cpu_info = subprocess.run(['nproc'], stdout=subprocess.PIPE).stdout
+    return int(cpu_info)
+
+def monitor_pipeline_pidstat(num_proc, data_description, output_parquet_path, output_results_path):
+    """
+    Creates output subdirs and executes bash pipeline monitoring script.
+
+    Args:
+      num_proc: Number of processes the batch job is set to run with
+      data_description: Description of the dataset
+      output_parquet_path: Directory of the output parquet files
+      output_results_path: Directory of the output csv files and graphs
+
+    Returns:
+      None
+    """
+    if not os.path.exists(output_results_path + str(data_description) + '/graphs'): os.makedirs(output_results_path + str(data_description) + '/graphs')
+    if not os.path.exists(output_results_path + str(data_description) + '/tables'): os.makedirs(output_results_path + str(data_description) + '/tables')
+    if not os.path.exists(output_results_path + str(data_description) + '/raw'): os.makedirs(output_results_path + str(data_description) + '/raw')
+    subprocess.call(['sh', './monitor_pipeline.sh', num_proc, data_description, output_parquet_path, output_results_path])
+
+def make_pidstat_tables(file_name, type):
+    """
+    Creates pandas tables from raw pidstat data.
+
+    Args:
+      file_name: Name of the .txt file containing raw pidstat output
+      type: Type of the resource relevant to the table (Pipeline, Server, or DB)
+
+    Returns:
+      Pandas table with CPU, memory and I/O usage info for the resource.
+    """
+
+    #Parse the raw .txt file into a pandas dataframe
+    total_usage = []
+    seconds_elapsed = 0
+    with open(file_name) as f:
+        for line in f:
+            usage = []
+            if line[0].isdigit():
+                usage = line.split()
+
+                if type == 'DB ' and usage[-3] != 'hapi':
+                    continue
+
+                del usage[:8]
+                del usage[9:]
+                del usage[1:6]
+                usage.insert(0, seconds_elapsed)
+                total_usage.append(usage)
+            elif line[0] == '#':
+                seconds_elapsed += 1
+                
+    total_usage_df = pd.DataFrame(total_usage, columns = ['Seconds Elapsed', type + '%CPU', type + 'MEM', type + 'kBs Read / Second', type + 'kBs Written / Second'])
```

Comment:
nit: I think it is nicer to use an enum for the `type`, but if you want to keep it as a string, please remove the trailing space (e.g., use `'DB'` instead of `'DB '`).